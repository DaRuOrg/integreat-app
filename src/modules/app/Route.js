import UrlPattern from 'url-pattern'

class Route {
  id
  path
  pattern

  constructor (id, path) {
    this.id = id
    this.path = path

    this.pattern = new UrlPattern(path)
  }

  stringify (params = {}) {
    return this.pattern.stringify(params)
  }

  hasPath (path) {
    return this.path === path
  }
}

export default Route
