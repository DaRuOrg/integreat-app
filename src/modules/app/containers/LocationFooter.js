import React from 'react'
import PropTypes from 'prop-types'
import { translate } from 'react-i18next'
import { connect } from 'react-redux'
import compose from 'lodash/fp/compose'

import Navigation from 'modules/app/Navigation'
import Footer from '../../layout/components/Footer'

class LocationFooter extends React.Component {
  static propTypes = {
    navigation: PropTypes.instanceOf(Navigation).isRequired
  }

  render () {
    const {t, navigation} = this.props
    return <Footer items={[{text: t('imprintAndContact'), href: navigation.disclaimer}]} />
  }
}

const mapStateToProps = (state) => ({
  navigation: new Navigation(state.router.params.location, state.router.params.language)
})

export default compose(
  connect(mapStateToProps),
  translate('app')
)(LocationFooter)
