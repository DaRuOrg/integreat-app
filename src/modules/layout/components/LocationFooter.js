import React from 'react'
import PropTypes from 'prop-types'
import { translate } from 'react-i18next'

import Navigation from 'modules/app/Navigation'
import Footer from './Footer'
import { Link } from 'redux-little-router'

class LocationFooter extends React.Component {
  static propTypes = {
    navigation: PropTypes.instanceOf(Navigation).isRequired,
    t: PropTypes.func.isRequired
  }

  render () {
    return <Footer>
      <Link href={this.props.navigation.disclaimer}>{this.props.t('imprintAndContact')}</Link>
      //todo translate Datenschutz
      <a href={'https://integreat-app.de/datenschutz/'}>Datenschutz</a>
    </Footer>
  }
}

export default translate('app')(LocationFooter)
