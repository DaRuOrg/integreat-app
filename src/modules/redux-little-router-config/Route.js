import UrlPattern from 'url-pattern'

class Route {
  id
  path
  pattern
  renderComponent
  condition

  constructor (id, path, renderComponent, condition) {
    this.id = id
    this.path = path
    this.renderComponent = renderComponent
    this.condition = condition

    this.pattern = new UrlPattern(path)
  }

  stringify (params = {}) {
    return this.pattern.stringify(params)
  }
}

export default Route
