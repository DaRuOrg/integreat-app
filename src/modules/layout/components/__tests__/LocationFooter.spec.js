import { shallow } from 'enzyme'
import React from 'react'
import LocationFooter from '../LocationFooter'
import Route from '../../../app/Route'

describe('LocationFooter', () => {
  test('should match snapshot', () => {
    const component = shallow(<LocationFooter
      currentParams={{
        location: 'augsburg',
        language: 'de'
      }}
      matchRoute={(id) => new Route(id, '/:location/:language/disclaimer')} />)
    expect(component.dive()).toMatchSnapshot()
  })
})
