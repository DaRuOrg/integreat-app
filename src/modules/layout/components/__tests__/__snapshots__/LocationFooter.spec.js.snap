// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationFooter should match snapshot 1`] = `
<Footer>
  <Connect(Link)
    href="/augsburg/de/disclaimer"
  >
    app:imprintAndContact
  </Connect(Link)>
  <a
    href="https://integreat-app.de/datenschutz/"
  >
    app:privacy
  </a>
</Footer>
`;
